test/FunctionalTests/Microsoft.OData.Core.Tests/ODataMessageWriterTests.cs

```diff
@@ -914,6 +967,42 @@ public async Task WriteMetadataDocumentAsync_WorksForXmlCsdl()
                 "</edmx:Edmx>", payload);
         }
 
+        [Fact]
+        public async Task WriteLargeMetadataDocumentAsync_MultipleTimes_WorksForXmlCsdl()
+        {
+            // Arrange
+            IEdmModel largeModel = GetLargeEdmModel();
+
+            var expectedPayload = await WriteAndGetPayloadAsync(largeModel, "application/xml", async writer =>
+            {
+                await writer.WriteMetadataDocumentAsync();
+            });
+
+            // Act - XML
+            for (int i = 0; i < 1000; i++)
+            {
+                string payload = await this.WriteAndGetPayloadAsync(largeModel, "application/xml", async omWriter =>
+                {
+                    try

```

Comment:
Why still have the try {} catch {}?
If throw exception, make sure it throws correctly.
If not, remove try/catch

Reply:
Removed